addons/website_slides/static/src/js/public/components/slide_upload_dialog/slide_upload_dialog.js

```diff
@@ -0,0 +1,123 @@
+/** @odoo-module **/
+
+import { Component, onMounted, useState } from "@odoo/owl";
+import { Dialog } from "@web/core/dialog/dialog";
+import { DropdownItem } from "@web/core/dropdown/dropdown_item";
+import { redirect } from "@web/core/utils/urls";
+import { SelectMenu } from "@web/core/select_menu/select_menu";
+import { ModuleToInstallIcon, SlideCategoryIcon } from "./slide_upload_dialog_select";
+import { SlideInstallModule } from "./slide_install_module";
+import { SlideUploadCategory } from "./slide_upload_category";
+import { _t } from "@web/core/l10n/translation";
+import { useService } from "@web/core/utils/hooks";
+
+export class SlideUploadDialog extends Component {
+    static baseSettings = {
+        modulesToInstallMsg: "",
+        page: "select",
+        size: "md",
+        alertMsg: "",
+        title: _t("Add Content"),
+        installModuleData: null,
+    };
+    static categoryData = {
+        document: { icon: "fa-file-pdf-o", label: _t("Document") },
+        infographic: { icon: "fa-file-image-o", label: _t("Image") },
+        article: { icon: "fa-file-text", label: _t("Article") },
+        video: { icon: "fa-file-video-o", label: _t("Video") },
+        quiz: { icon: "fa-question-circle", label: _t("Quiz") },
+    };
+    static components = {
+        Dialog,
+        DropdownItem,
+        ModuleToInstallIcon,
+        SelectMenu,
+        SlideCategoryIcon,
+        SlideUploadCategory,
+        SlideInstallModule,
+    };
+    static pagesTemplates = {
+        article: "website_slides.SlideCategoryTutorial.Article",
+        document: "website_slides.SlideCategoryTutorial.Document",
+        infographic: "website_slides.SlideCategoryTutorial.Infographic",
+        select: "website_slides.SlideUploadDialogSelect",
+        install_module: "website_slides.UploadDialogInstallModule",
+        upload: "website_slides.UploadInProgressDialog",
+        video: "website_slides.SlideCategoryTutorial.Video",
+        quiz: "website_slides.SlideCategoryTutorial.Quiz",
+    };
+    static props = {
+        canPublish: Boolean,
+        canUpload: Boolean,
+        categoryId: { type: String, optional: true },
+        channelId: Number,
+        close: Function,
+        modulesToInstall: { type: Array, optional: true },
+        openModal: { type: String, optional: true },
+    };
+    static template = "website_slides.SlideUploadDialog";
+
+    setup() {
+        super.setup();
+        this.defaultCategoryID = parseInt(this.props.categoryId, 10);
+        this.modulesToInstallStatus = null;
+        this.dialog = useService("dialog");
+        this.orm = useService("orm");
+        this.rpc = useService("rpc");
+        this.pagesTemplates = SlideUploadDialog.pagesTemplates;
```

Comment:
You maybe want to use this.constructor.pagesTemplates ? (with this solution you can overwrite it when you extend your component)